### PR TITLE
JAMES-3775 Refactor GetMailboxMessagesService + Clean code + Fix webadmin task

### PR DESCRIPTION
**Pull request by vttranlina**
- Avoid collectList

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java [hunk 1]
```diff
@@ -90,9 +90,7 @@ private Flux<MessageResult> getMailboxMessagesOfAUser(Username username, String
             .doOnNext(mailboxMessageMetaData -> context.incrementSpamMessageCount())
             .filter(mailboxMessageMetaData -> afterDate.map(date -> mailboxMessageMetaData.getInternalDate().after(date)).orElse(true))
             .filter(message -> randomBooleanWithProbability(runningOptions))
-            .map(Message::getMessageId)
-            .collectList()
-            .flatMapMany(messageIds -> messageIdManager.getMessagesReactive(messageIds, FetchGroup.FULL_CONTENT, mailboxSession))
+            .flatMap(message -> messageIdManager.getMessagesReactive(List.of(message.getMessageId()), FetchGroup.FULL_CONTENT, mailboxSession))
```
Comment (chibenwa) [inline third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:93]: We likely can use window operator to still read msgs by batch without reading them all

Comment (vttranlina) [reply third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:93]: "window" is expensive, should we still consider in this case?

Comment (chibenwa) [reply third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:93]: No window by item count is cheap, but likely not necessary perf wise. Let's go ahead imo

**Review by chibenwa — APPROVED** [review]

File: third-party/rspamd/src/test/java/org/apache/james/rspamd/client/RspamdHttpClientTest.java [hunk 2]
```diff
@@ -95,24 +115,24 @@ void learnHamWithWrongPasswordShouldThrowUnauthorizedExceptionException() {
...
-        AnalysisResult analysisResult = client.checkV2(new ByteArrayInputStream(spamMessage)).block();
+        AnalysisResult analysisResult = client.checkV2(spamMessage).block();
         assertThat(analysisResult.getAction()).isEqualTo(AnalysisResult.Action.REJECT);
 
         RequestSpecification rspamdApi = WebAdminUtils.spec(Port.of(rspamdExtension.dockerRspamd().getPort()));
         rspamdApi
             .header(new Header("Password", PASSWORD))
             .body(ClassLoader.getSystemResourceAsStream(SPAM_MESSAGE_PATH))
-            .post("checkv2")
+            .post("checkv2").prettyPeek()
```
Comment (chibenwa) [inline third-party/rspamd/src/test/java/org/apache/james/rspamd/client/RspamdHttpClientTest.java:135]: pretty peek

File: third-party/rspamd/src/test/java/org/apache/james/rspamd/client/RspamdHttpClientTest.java [hunk 3]
```diff
@@ -95,24 +115,24 @@ void learnHamWithWrongPasswordShouldThrowUnauthorizedExceptionException() {
...
 
-        assertThatThrownBy(() -> reportAsHam(client, new ByteArrayInputStream(spamMessage)))
+        assertThatThrownBy(() -> reportAsHam(client, spamMessage.getMessage().getInputStream()))
             .hasMessage("{\"error\":\"Unauthorized\"}")
             .isInstanceOf(UnauthorizedException.class);
     }
 
     @Test
-    void checkSpamMailUsingRspamdClientWithExactPasswordShouldReturnAnalysisResultAsSameAsUsingRawClient() {
+    void checkSpamMailUsingRspamdClientWithExactPasswordShouldReturnAnalysisResultAsSameAsUsingRawClient() throws MessagingException {
```
Comment (chibenwa) [inline third-party/rspamd/src/test/java/org/apache/james/rspamd/client/RspamdHttpClientTest.java:124]: throws Exception

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
And needs a rebase

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**